### PR TITLE
Rename ST_CENTROID to ST_CENTROID_AGG

### PR DESCRIPTION
**Pull request by craigtaverner**
In order to allow development of a scalar ST_CENTROID function.

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public class EsqlFeatures implements FeatureSpecification {
     /**
      * Support for spatial aggregation {@code ST_CENTROID}. Done in #104269.
      */
-    private static final NodeFeature ST_CENTROID = new NodeFeature("esql.st_centroid");
+    private static final NodeFeature ST_CENTROID_AGG = new NodeFeature("esql.st_centroid_agg");
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:71]: Good thing we marked that as experimental in the docs and can just make breaking changes!

**Review by alex-spies — APPROVED** [review]
LGTM!

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/meta.csv-spec [hunk 2]
```diff
@@ -174,7 +174,7 @@ sin           |angle                               |"double|integer|long|unsigne
 sinh          |angle                               |"double|integer|long|unsigned_long"                                                                                               |An angle, in radians. If `null`, the function returns `null`.
 split         |[string, delim]                     |["keyword|text", "keyword|text"]                                                                                                  |[, ]
 sqrt          |number                              |"double|integer|long|unsigned_long"                                                                                               |[""]
-st_centroid   |field                               |"geo_point|cartesian_point"                                                                                                       |[""]
+st_centroid_ag|field                              |"geo_point|cartesian_point"                                                                                                       |[""]
```
[the remark this answers is not in the document]
Comment (bpintea) [reply x-pack/plugin/esql/qa/testFixtures/src/main/resources/meta.csv-spec:177]: (..and same below)

[the remark this answers is not in the document]
Comment (bpintea) [reply x-pack/plugin/esql/qa/testFixtures/src/main/resources/meta.csv-spec:177]: (..and same below)

**Review by bpintea — APPROVED** [review]

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/meta.csv-spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (craigtaverner) [reply x-pack/plugin/esql/qa/testFixtures/src/main/resources/meta.csv-spec:177]: This is not the only function truncated at this point, so thought it best to follow the pattern and keep the truncation.

[the remark this answers is not in the document]
Comment (craigtaverner) [reply x-pack/plugin/esql/qa/testFixtures/src/main/resources/meta.csv-spec]: But I'll fix the table alignment by adding extra space before the next column.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java (hunk 1 above)
Comment (craigtaverner) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:71]: Personally I hope in future to allow the same name to work as both scalar and aggregating functions, but by then we'll need to go through a deprecation phase to remove the _AGG version. Renaming now gives us more options going forward, with less bwc issues.

**Pull request merged**